### ISSUE TITLE
Allow gcode-modifiers to not split areas.

### ISSUE THREAD

**Issue by CCS86** [supermerill/SuperSlicer#914]
2.3.55.2 / Win 10 x64

It seems as though bridge perimeters are not being considered "bridges", and are causing the infill area above to split, with a non-100% above-bridge flow ratio:

![image](https://user-images.githubusercontent.com/12439543/105638807-69bc6300-5e3a-11eb-8133-0ec7902c813b.png)

![image](https://user-images.githubusercontent.com/12439543/105638810-6d4fea00-5e3a-11eb-8652-ad015e05579e.png)

[Bridge Test.3mf.txt](https://github.com/supermerill/SuperSlicer/files/5862843/Bridge.Test.3mf.txt)

IMO, "bridges" and "bridging perimeters" should be considered the same for this purpose and not split.

**Comment by qwewer0** [comment]
The issue is present on 2.3.55.4.

To reproduce the issue, use `over_bridge_flow_ratio != 100%`,
and for better visualization use `only_one_perimeter_top = 1` with `perimeters > 1`

**Comment by CCS86** [comment]
> The issue is present on 2.3.55.4.
> 
> To reproduce the issue, use `over_bridge_flow_ratio != 100%`,
> and for better visualization use `only_one_perimeter_top = 1` with `perimeters > 1`



2.3.55.4?

I don't see that in releases.

**Comment by qwewer0** [comment]
@CCS86 You can find it in the [Actions](https://github.com/supermerill/SuperSlicer/actions) tab

**Comment by supermerill** [comment]
That's because of the over_bridge_flow_ratio. That is used to compensate a bit the drop from the bridge area.
You can set it to 100% to remove this split when it's not needed.

**Comment by supermerill** [comment]
mmm. Maybe the problem is that the over_bridge_flow_ratio should also be used over the overhangs ?

**Comment by qwewer0** [comment]
> mmm. Maybe the problem is that the over_bridge_flow_ratio should also be used over the overhangs ?

@supermerill That might be a good idea, but I think this issues main point it that the over_bridge_flow_ratio line isn't printed continuously with the other lines.

**Comment by CCS86** [comment]
> mmm. Maybe the problem is that the over_bridge_flow_ratio should also be used over the overhangs ?

I would tend to say yes to this. But, I only use "overhang perimeters" as "bridge perimeters", with a 100% unsupported requirement. 

Maybe "bridge perimeters" and "overhang perimeters" need to be split into discrete categories? Then, bridges are bridges, whether perimeter or internal, and can be combined for these flow overrides.

Also, the "above bridge" flow override seems to do nothing (besides splitting the infill). Flow rate is not increased at all. I can start a new issue if you would like.

**Comment by supermerill** [comment]
If I remember, it doesn't increase flow rate but creates more overlap, so it extrudes more plastic in the same area.

**Comment by qwewer0** [comment]
I think this issue is still present on the latest release.

over_bridge_flow_ratio: 99%
![bandicam 2021-06-04 19-14-11-867](https://user-images.githubusercontent.com/57561110/120839707-91020a00-c569-11eb-90bb-a7a7091566f0.jpg)

over_bridge_flow_ratio: 100%
![bandicam 2021-06-04 19-13-41-464](https://user-images.githubusercontent.com/57561110/120839719-93646400-c569-11eb-8d36-1069e457433b.jpg)

**Comment by supermerill** [comment]
there is already issue #234 to remind me of it.

**Comment by supermerill** [comment]
> I think this issue is still present on the latest release.

it's something I wanted, that's the whole point of the feature.
This can only go away when modifiers will stop to create split in perimeters & infill.

**Comment by qwewer0** [comment]
> it's something I wanted, that's the whole point of the feature. Use 3 top layers or no over-bridge flow (100%) to don't see it.
> This can only go away when modifiers will stop to create split in perimeters & infill.

Yes, but as we now know why this happens, then could we see https://github.com/supermerill/SuperSlicer/issues/914 as a feature request?

**Comment by supermerill** [comment]
if you want.
there is also #1092 that is the same thing but for perimeters.

**Comment by qwewer0** [comment]
> if you want.
> there is also #1092 that is the same thing but for perimeters.

Thank you, very kind of you.
